.github/workflows/static_analysis.yml

```diff
@@ -17,13 +17,14 @@ concurrency:
 
 env:
   COMPILER: 'gcc'
+  MAKEFLAGS: '-j4'
   OS_NAME: 'linux'
   PYTHON_VERSION: '3.12'
 
 jobs:
   test:
     name: ${{ matrix.task }}
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```

Comment:
This will prevent needing to update this OS version in the future. I don't think there's any reason this job needs to be on 22.04 specifically.